### PR TITLE
Invisible Foam Force Shotgun and Double Barrel Shotguns Fix

### LINKED ISSUE #11864: Double barrel shotgun alt sprites and sawn-off sprite are invisible
Issue reported from  Round ID: 42349 (Hippie Station)

Reporting client version: 512

what it says on the tin quite honestly; Alternate skins for the double barrel shotgun and the sawn-off double barrel shotgun do not have sprites

### PR DESCRIPTION
**Pull request by IHadMyUsernameTaken**
## Changelog
:cl:
fix: Foam force shotguns are now visible.
fix: Alternative skins of the double barrel shotgun and the sawn-off double barrel shotgun no longer are invisible.
/:cl:
## About The Pull Request
Foam force shotguns were invisible, because they weren't looking at the toy.dmi file where the shotgun icon state was.

Double barrel shotguns and sawn-off double barrel shotguns didn't have any visible alternative skins due to the icon states in the projectile.dmi being named so that the possible icon states in the unique_reskin list for a double barrel shotgun weren't valid. Fixes #11864.
As for why I renamed the "_sawn" suffix as "-sawn", none of the possible alternative skins of the sawn-off double barrel shotgun were using the "_sawn" suffix. You could also make an invalid icon_state "dshotgun_sawn" in this case if you happened to choose the Default skin for a double barrel shotgun and then saw its barrel to make it into a sawn-off double barrel shotgun.

## Why It's Good For The Game
Having these use their icon states is probably better than them being invisible.

### CODE REVIEW

**Comment by Manly-Dwarf** [comment]
who is this mysterious legend fixing all of our bugs

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -72,7 +72,7 @@
 		return
 	..()
 	if(current_skin)
-		icon_state = "[unique_reskin[current_skin]][sawn_off ? "_sawn" : ""]"
+		icon_state = "[unique_reskin[current_skin]][sawn_off ? "-sawn" : ""]"
```
Comment (Manly-Dwarf) [inline code/modules/projectiles/guns/ballistic.dm]:
```suggestion
		icon_state = "[unique_reskin[current_skin]][sawn_off ? "-sawn" : ""] // hippie -- fixes sawn-off sprites"
```

**Review by Manly-Dwarf — CHANGES_REQUESTED** [review]
We're not synced up to the mirror bot, but just in case SOMEONE decides to do an upstream merge, it's best to still mark changes you made to the original files. Just in case.

**Review by Manly-Dwarf — DISMISSED** [review]
:heck:

File: code/modules/projectiles/guns/ballistic.dm [hunk 2]
```diff
@@ -72,7 +72,7 @@
 		return
 	..()
 	if(current_skin)
-		icon_state = "[unique_reskin[current_skin]][sawn_off ? "_sawn" : ""]"
+		icon_state = "[unique_reskin[current_skin]][sawn_off ? "-sawn" : ""] // hippie -- fixes sawn-off sprites"
```
Comment (Manly-Dwarf) [inline code/modules/projectiles/guns/ballistic.dm]:
```suggestion
		icon_state = "[unique_reskin[current_skin]][sawn_off ? "-sawn" : ""]" // hippie -- fixes sawn-off sprites
```

**Review by Manly-Dwarf — CHANGES_REQUESTED** [review]
oh fuck wait ABORT ABORT

File: code/modules/projectiles/guns/ballistic.dm (hunk 2 above)
Comment (IHadMyUsernameTaken) [reply code/modules/projectiles/guns/ballistic.dm]: Oh damn, I should've looked, good stuff.

**Review by Manly-Dwarf — APPROVED** [review]
:heck:

**Comment by yoyobatty** [comment]
Thanks, I guess I missed this one.

**Pull request merged**